nix-command and gerbsup updates

AlexKnauth opened the pull request.

AlexKnauth committed:
nix --extra-experimental-features nix-command

fare commented on scripts/update-gerbil-nix-recipe.ss:
```diff
@@ -47,22 +50,42 @@
   (def recipe-file (path-expand recipe-path nixpkgs-dir))
 
+  (unless (file-exists? source-dir)
+    (error/dir-not-found name source-dir (filter (cut string-contains <> name) cmdopts)))
```
I don't understand the string-contains logic—is that a debug tool to check that the options are well-named? That deserves a comment.

AlexKnauth replied:
I was thinking the `cmdopts` might contain options that apply to multiple things: the source-dir, the checkout-dir, and the nixpkgs-dir. I ended up not doing that, so I can change it back and clarify.

fare commented on scripts/update-gerbil-nix-recipe.ss:
```diff
@@ -15,6 +15,9 @@
 (defonce (default-checkouts-dir)
   (path-maybe-normalize (path-expand "../.." (path-directory (this-source-file)))))
 
```
I propose you use the new `(this-source-directory)`

AlexKnauth replied:
Oh, thanks, I'll do that.

fare commented on scripts/update-gerbil-nix-recipe.ss:
```diff
@@ -274,7 +307,8 @@
       nixpkgs-dir: nixpkgs-dir
-      stable: stable))
+      stable: stable
+      cmdopts: ["--smug-gerbil-dir" "--gerbil-off" "--stable"]))
```
While we're at it, shouldn't we uniformly add the -repo and -dir arguments everywhere?

AlexKnauth replied:
Sure I can do that.

fare reviewed (approved):
LGTM

AlexKnauth committed:
update gerbsup: better error messages and checkouts-by-owner fallback

AlexKnauth commented:
Okay, I have changed those three things:
 * Used `(this-source-directory)`,
 * Removed filter of `cmdopts` and added a comment to clarify its meaning,
 * Added `-repo` command-line options for everything that didn't have them already to be consistent.
